build/media_source/system/js/fields/joomla-media-select.w-c.es6.js

```diff
@@ -317,7 +317,14 @@ Joomla.getMedia = (data, editor, fieldClass) => new Promise((resolve, reject) =>
     return;
   }
 
-  const url = `${Joomla.getOptions('system.paths').baseFull}index.php?option=com_media&task=api.files&url=true&path=${encodeURIComponent(data.path)}&mediatypes=0,1,2,3&${Joomla.getOptions('csrf.token')}=1&format=json`;
+  // Compile the url
+  const url = new URL(Joomla.getOptions('media-picker-api').apiBaseUrl ? Joomla.getOptions('media-picker-api').apiBaseUrl : Joomla.getOptions('system.paths').baseFull + 'index.php?option=com_media&format=json');
+  url.searchParams.append('task', 'api.files');
+  url.searchParams.append('url', true);
+  url.searchParams.append('path', data.path);
```

Comment:
```suggestion
  url.searchParams.append('path', encodeURIComponent(data.path));
```

Please keep the `encodeURIComponent` as it seems that the browsers don't do that automatically: https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams#preserving_plus_signs